Dev4Agriculture.ISO11783.ISOXML/Geometry/GeometryUtility.cs

```diff
@@ -0,0 +1,222 @@
+            {
+                var firstPoint = points[i];
+                var checkPoint = points[i + 1];
+                var secondPoint = points[i + 2];
+                if (PointOnLineSegment(firstPoint, secondPoint,checkPoint))
+                {
+                    points.Remove(checkPoint);
+                    i = 0;
+                }
+            }
+            return points;
+        }
+
+        private static bool PointOnLineSegment(ISOPoint pt1, ISOPoint pt2, ISOPoint pt, decimal epsilon = 0.0001m)
+        {
+            if (pt.PointNorth - Math.Max(pt1.PointNorth, pt2.PointNorth) > epsilon || 
+                Math.Min(pt1.PointNorth, pt2.PointNorth) - pt.PointNorth > epsilon || 
+                pt.PointEast - Math.Max(pt1.PointEast, pt2.PointEast) > epsilon || 
+                Math.Min(pt1.PointEast, pt2.PointEast) - pt.PointEast > epsilon)
+                return false;
+
+            if (Math.Abs(pt2.PointNorth - pt1.PointNorth) < epsilon)
+                return Math.Abs(pt1.PointNorth - pt.PointNorth) < epsilon || Math.Abs(pt2.PointNorth - pt.PointNorth) < epsilon;
+            if (Math.Abs(pt2.PointEast - pt1.PointEast) < epsilon)
+                return Math.Abs(pt1.PointEast - pt.PointEast) < epsilon || Math.Abs(pt2.PointEast - pt.PointEast) < epsilon;
+
+            var x = pt1.PointNorth + (pt.PointEast - pt1.PointEast) * (pt2.PointNorth - pt1.PointNorth) / (pt2.PointEast - pt1.PointEast);
+            var y = pt1.PointEast + (pt.PointNorth - pt1.PointNorth) * (pt2.PointEast - pt1.PointEast) / (pt2.PointNorth - pt1.PointNorth);
+
+            return Math.Abs(pt.PointNorth - x) < epsilon || Math.Abs(pt.PointEast - y) < epsilon;
+        }
+
+        public static bool IsPolygonsEqual(List<ISOPoint> polygon1, List<ISOPoint> polygon2)
+        {
+            if (polygon1.Count != polygon2.Count)
+            {
+                return false;
+            }
+            for (var i = 0; i < polygon1.Count; i++)
+            {
+                var p1 = polygon1[i];
+                var p2 = polygon2[i];
+                if (p1.PointNorth != p2.PointNorth || p1.PointEast != p2.PointEast)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        private static List<ISOPoint> OrderPointsByDistanceFromCenter(this List<ISOPoint> points)
+        {
+            if (!points.Any())
+                return points;
+            decimal mX = 0;
+            decimal my = 0;
+            foreach (var p in points)
+            {
+                mX += p.PointEast;
+                my += p.PointNorth;
+            }
+            mX /= points.Count();
+            my /= points.Count();
+            return points.OrderBy(v => Math.Atan2((double)(v.PointNorth - my), (double)(v.PointEast - mX))).ToList();
+        }
+
+        public static List<ISOPoint> GetIntersectionOfPolygons(this List<ISOPoint> poly1, List<ISOPoint> poly2)
```

Comment:
Could you please add a function description here, mentioning the limits of this function (e.g. compatibility with concave/convex) ?